### ISSUE TITLE
Accept suggestion on commit character doesn't work inside function call

### ISSUE THREAD

**Issue by benasvip** [microsoft/vscode#168248]
Type: <b>Bug</b>

VS Code version: Code 1.73.1 (6261075646f055b99068d3688932416f2346dd3b, 2022-11-09T02:22:48.959Z)
OS version: Darwin arm64 22.1.0

Accept suggestion on commit character doesn't work inside function call, but works if called outside function call. It seems to unable to handle parentheses, because if it's brackets it works fine. Below you will find code snippet and gif recordings which reproduces the issue.

Setting: `"editor.acceptSuggestionOnCommitCharacter": true`

<details>
<summary><b>Code snippets</b></summary>

```ts
// | - means cursor position

const obj = {
  name: 'VS Code'
}

// 🟢 Works fine if outside of function call.
// ℹ️ How to reproduce: Write "o" letter and press `.` it completes to `obj.|`
obj.|

// 🟢 Also, works fine with object brackets.
console.log({ name: obj.| })

// 🟢 Works fine if "obj" was already there and I wanted to use "name" property.
// ℹ️ How to reproduce: Write "obj" word and press `.` it shows "name" and press `.` again.
console.log(obj.name.|)

// 🔴 Actual result: o.|
// Expected result: obj.|
// ℹ️ How to reproduce: Write letter "o" inside `console.log(|)` call and press `.`
console.log(o.|)
```
</details>

<details>
<summary><b>GIF recordings</b></summary>

**🟢 Works**

![works-1](https://user-images.githubusercontent.com/10441177/206050051-2f9693aa-35df-467b-a997-7f33ba30ec40.gif)
![works-2](https://user-images.githubusercontent.com/10441177/206050044-03935390-7ad6-4557-9a70-dbee72546e65.gif)
![works-3](https://user-images.githubusercontent.com/10441177/206050038-5577ffd5-6d01-4bef-a174-6c803f55bd61.gif)

**🔴 Doesn't work**

![not-working](https://user-images.githubusercontent.com/10441177/206050095-4527bb19-47c7-4e05-b490-dd3f145d0921.gif)

</details>

**Comment by svipas** [comment]
I believe @jrieken is the owner of suggestions correct me if I'm wrong.

**Comment by svipas** [comment]
Also, what is weird is that it works if `obj` was already completed before:

```ts
// 🟢 Works fine if "obj" was already there and I wanted to use "name" property.
// ℹ️ How to reproduce: Write "obj" word and press `.` it shows "name" and press `.` again.
console.log(obj.name.|)
```

**Comment by t7yang** [comment]
is this a related issue?
it just happens recently on insiders

![2022-12-09 17 32 33](https://user-images.githubusercontent.com/4502661/206670832-b8679b62-8382-4453-97ae-cbcd4be1e12a.gif)

Version: 1.75.0-insider (user setup)
Commit: b92894493e6f971a3c71912d823e699d3a05d643
Date: 2022-12-08T05:20:51.149Z
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19045
Sandboxed: Yes

**Comment by svipas** [comment]
@t7yang This is a separate issue, I suggest to create one and include all this info.

**Comment by benasvip** [comment]
@mjbvz This is now fixed with 5.7.0-dev.20240823 and as I read in TS blog this should be available in TS 5.6.
